Remotion/Web/ClientScript/Scripts/DropDownMenu.ts

```diff
@@ -0,0 +1,934 @@
+
+    var ul = document.createElement('ul');
+    ul.className = 'DropDownMenuOptions';
+    ul.setAttribute('role', 'none');
+    div.appendChild(ul);
+
+    const $menuButton = $('#' + menuID);
+    $menuButton.closest('div, td, th, body').append(div);
+
+    $(ul).mouseover (function (event)
+    {
+      var eventTarget = DropDownMenu.GetTarget (event, "LI");
+      $("li", ul).removeClass(DropDownMenu._itemSelectedClassName);
+      if (eventTarget.firstChild != null && eventTarget.firstChild.nodeName.toLowerCase() === 'a')
+      {
+        $(eventTarget).addClass (DropDownMenu._itemSelectedClassName);
+        (eventTarget.firstChild as HTMLElement).focus();
+      }
+    }).keydown (function (event)
+    {
+      // TODO: missing nullcheck?
+      DropDownMenu.Options_OnKeyDown (event, DropDownMenu._currentMenu!);
+    });
+
+    DropDownMenu._itemClickHandler = function(this: HTMLAnchorElement)
+    {
+      if (this.href == null || this.href === '')
+        return false;
+
+      DropDownMenu._itemClicked = true;
+      DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+      try
+      {
+        eval(this.getAttribute('javascript')!);
+      }
+      catch (e)
+      {
+      }
+      setTimeout (function () { DropDownMenu._itemClicked = false; }, 10);
+      return false;
+    };
+
+    for (let i = 0; i < itemInfos.length; i++)
+    {
+      var item = DropDownMenu.CreateItem(itemInfos[i], selectionCount);
+      if (item != null)
+        ul.appendChild(item);
+    }
+
+    var titleDivFunc = DropDownMenu.CreateTitleDivGetter (context);
+    DropDownMenu.ApplyPosition ($(div), evt, titleDivFunc());
+    $(div).iFrameShim({ top: '0px', left: '0px', width: '100%', height: '100%' });
+
+    if (DropDownMenu._repositionTimer) 
+      clearTimeout(DropDownMenu._repositionTimer);
+    var repositionHandler = function ()
+    {
+      if (DropDownMenu._repositionTimer)
+        clearTimeout (DropDownMenu._repositionTimer);
+
+      if (DropDownMenu._currentPopup && DropDownMenu._currentPopup == div && $(div).is (':visible'))
+      {
+        DropDownMenu.ApplyPosition ($(div), null, titleDivFunc());
+        DropDownMenu._repositionTimer = setTimeout (repositionHandler, DropDownMenu._repositionInterval);
+      }
+    };
+
+    // Only reposition if opened via titleDiv
+    if (evt == null)
+      DropDownMenu._repositionTimer = setTimeout(repositionHandler, DropDownMenu._repositionInterval);
+
+    if ($menuButton.data(DropDownMenu._button_timestampDataKey))
+    {
+      $menuButton.removeData(DropDownMenu._button_timestampDataKey);
+    }
+    else
+    {
+      DropDownMenu.ClosePopUp(DropDownMenu._updateFocus);
+    }
+  }
+
+  private static CreateTitleDivGetter ($context: JQuery): DropDownMenu_TitleDivGetter
+  {
+    const context = $context[0];
+    const contextID = context.id;
+    if (contextID == null)
+    {
+      return function ()
+      {
+        // TODO: nullcheck?
+        return $(context.firstElementChild!);
```

Comment:
how did we deal with not-supported dom-structures in other files? just ignore or add a dedicated exception?

Reply:
Sorry, I don't understand

Reply:
We already have precedent for the case when the DOM rendered by us does not match the Javascript and would thus trigger a null-exception. @a-ctor should know where we found those cases. I want to deal with these cases the same way whenever they pop up.

Reply:
We have been ignoring cases as it would not make much sense to validate the DOM structure with each call. Ideally the JavaScript would create all the DOM element and store them in the class. This way we wouldn't have to worry about the DOM structure and simply access the elements through the class members.

Reply:
@a-ctor regarding "ideally" => only when viewing JavaScript alone. We don't do an SPA here, the JavaScript just extends the rendered features. To move "radnom" stuff into JavaScript would make for more complicted client/server interaction and it would also ad to the startup-load of the page.

Reply:
What's the decision here? Remove the `TODO` + leave `!`?

Reply:
Result of a discussion with MK: Yes, remove `TODO` and leave with a BANG